CMakeLists.txt

```diff
@@ -171,8 +171,16 @@ if(USE_CUDA)
     set(CUDA_ARCHS "6.0" "6.1" "6.2" "7.0")
     if(CUDA_VERSION VERSION_GREATER_EQUAL "10.0")
         list(APPEND CUDA_ARCHS "7.5")
+    elseif(CUDA_VERSION VERSION_GREATER_EQUAL "11.0")
+        list(APPEND CUDA_ARCHS "8.0")
     endif()
     list(POP_BACK CUDA_ARCHS CUDA_LAST_SUPPORTED_ARCH)
+    foreach(arch ${CUDA_ARCHS})
+      string(REPLACE "." "" ver ${arch})
+      set(CUDA_ARCH_VERSION "${ver}-real;${ver}-virtual;${CUDA_ARCH_VERSION}")
+    endforeach()
```

Comment:
Can we simplify the whole loop to two lines according to the https://cmake.org/cmake/help/latest/policy/CMP0104.html#examples?

```
# convert CUDA_ARCHS to string (let say CUDA_ARCHS_STRING) and remove from it all dots (.)
set_property(TARGET tgt PROPERTY CUDA_ARCHITECTURES CUDA_ARCHS_STRING)
```